aws-sns-topicpolicy/src/main/java/software/amazon/sns/topicpolicy/BaseHandlerStd.java

```diff
@@ -1,204 +1,170 @@
+      }
+    } else {
+      ex = new CfnGeneralServiceException(e);
     }
-
-    /**
-     * Invocation of getPolicyDocument returns the policy document .
-     *
-     * @param request
-     *            {@link ResourceHandlerRequest<ResourceModel>}
-     * @return Returns policy document
-     */
-    private String getPolicyDocument(final ResourceHandlerRequest<ResourceModel> request) {
-        try {
-            return MAPPER.writeValueAsString(request.getDesiredResourceState().getPolicyDocument());
-        } catch (JsonProcessingException e) {
-            throw new CfnInvalidRequestException(e);
-        }
+    return ProgressEvent.failed(resourceModel, callbackContext, ex.getErrorCode(), ex.getMessage());
+  }
+
+
+  /**
+   * Invocation of getPolicyDocument returns the policy document .
+   *
+   * @param request
+   *            {@link ResourceHandlerRequest<ResourceModel>}
+   * @return Returns policy document
+   */
+  protected String getPolicyDocument(final ResourceHandlerRequest<ResourceModel> request) {
+    try {
+      return MAPPER.writeValueAsString(request.getDesiredResourceState().getPolicyDocument());
+    } catch (JsonProcessingException e) {
+      throw new CfnInvalidRequestException(e);
     }
+  }
+
+  /*
+   * Unfortunately, SNS requires a policy for a topic, so we must reset to the default policy for now. The original
+   * default policy, created by the SNS service (not CloudFormation) when the topic was created, specified an
+   * uppercase SNS: asin spite of the public AWS documentation always using a lowercase sns:. So this method uses SNS:
+   * as the service does. The Ruby integration tests will fail if the code below uses sns: instead.
+   */
+  static String getDefaultPolicy(final ResourceHandlerRequest<ResourceModel> request, String topicArn) {
+    String accountId = request.getAwsAccountId();
+    StringBuilder sb = new StringBuilder()
+            .append("{")
+            .append("    \"Version\": \"2008-10-17\",")
+            .append("    \"Id\": \"__default_policy_ID\",")
+            .append("    \"Statement\": [")
+            .append("      {")
+            .append("        \"Effect\": \"Allow\",")
+            .append("        \"Sid\": \"__default_statement_ID\",")
+            .append("        \"Principal\": {")
+            .append("          \"AWS\": \"*\"")
+            .append("        },")
+            .append("        \"Action\": [")
+            .append("          \"SNS:GetTopicAttributes\",")
+            .append("          \"SNS:SetTopicAttributes\",")
+            .append("          \"SNS:AddPermission\",")
+            .append("          \"SNS:RemovePermission\",")
+            .append("          \"SNS:DeleteTopic\",")
+            .append("          \"SNS:Subscribe\",")
+            .append("          \"SNS:ListSubscriptionsByTopic\",")
+            .append("          \"SNS:Publish\",")
+            .append("          \"SNS:Receive\"")
```

Comment:
This is a deprecated permission, we should remove this. We've had a CR in legacy workflow package to do the same change as well.

Reply:
I still find it in native code. So I should just remove it?

Reply:
Yes, we have a CR pending review, CR-87252060, you can go and check internally. Appreciated if you could approve that as well.